pkg/diff/sql_vertex_generator.go

```diff
@@ -0,0 +1,229 @@
+package diff
+
+import (
+	"fmt"
+
+	"github.com/stripe/pg-schema-diff/internal/schema"
+)
+
+type diffType string
+
+const (
+	diffTypeDelete   diffType = "DELETE"
+	diffTypeAddAlter diffType = "ADDALTER"
+)
+
+// partialSQLGraph represents the SQL statements and their dependencies. Every SQL statement is a
+// vertex in the graph. This is different from a graph because a dependency may exist between a node
+// within this part and a node in another part.
+type partialSQLGraph struct {
+	vertices     []sqlVertex
+	dependencies []dependency
+}
+
+func (s *partialSQLGraph) statements() []Statement {
+	var statements []Statement
+	for _, vertex := range s.vertices {
+		statements = append(statements, vertex.statements...)
+	}
+	return statements
+}
+
+func concatPartialGraphs(parts ...partialSQLGraph) partialSQLGraph {
+	var vertices []sqlVertex
+	var dependencies []dependency
+	for _, part := range parts {
+		vertices = append(vertices, part.vertices...)
+		dependencies = append(dependencies, part.dependencies...)
+	}
+	return partialSQLGraph{
+		vertices:     vertices,
+		dependencies: dependencies,
+	}
+}
+
+func graphFromPartials(parts partialSQLGraph) (*sqlGraph, error) {
+	graph := newSqlGraph()
+	for _, vertex := range parts.vertices {
+		// It's possible the node already exists. merge it if it does
+		if graph.HasVertexWithId(vertex.GetId()) {
+			vertex = mergeVertices(graph.GetVertex(vertex.GetId()), vertex)
+		}
+		graph.AddVertex(vertex)
+	}
+
+	for _, dep := range parts.dependencies {
+		sourceVertex := sqlVertex{
+			id:         dep.source,
+			priority:   sqlPriorityUnset,
+			statements: nil,
+		}
+		targetVertex := sqlVertex{
+			id:         dep.target,
+			priority:   sqlPriorityUnset,
+			statements: nil,
+		}
+
+		// To maintain the correctness of the graph, we will add a dummy vertex for the missing dependencies
```

Comment:
We probably want to remove this behavior in the future, but that's out-of-scope of this PR